### PR TITLE
Switch usage of process.EventEmitter (removed in Node 7) for require('events')

### PR DESCRIPTION
**Pull request by inversion**

### CODE REVIEW

**Comment by benburry** [comment]
Thanks!

**Comment by powellchristoph** [comment]
FYI, I haven't had time to debug it yet, but I am receiving this error now.

```

util.js:556
  ctor.prototype = Object.create(superCtor.prototype, {
                          ^
TypeError: Object prototype may only be an Object or null
    at Function.create (native)
    at Object.exports.inherits (util.js:556:27)
    at Object.<anonymous> (/usr/share/statsd/lib/config.js:33:6)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/usr/share/statsd/stats.js:4:14)
```

**Comment by benburry** [comment]
@powellchristoph could you tell me what version of Node you're running?

**Comment by powellchristoph** [comment]
I was running `master`.

**Comment by benburry** [comment]
Apologies for the oversight. This is fixed in 8d5363c

**Comment by powellchristoph** [comment]
Thanks!

**Comment by bashofmann** [comment]
@benburry It would be great to also tag a new release with this fix

**Comment by netroy** [comment]
@kastner @mrtazz @draco2003 @pathzzrd can we please have new version of the module published to npm? without this change, the module doesn't work on node-8

File: lib/config.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ var Configurator = function (file) {
   });
 };
 
-util.inherits(Configurator, process.EventEmitter);
+util.inherits(Configurator, require('events'));
```
Comment (pmmusial) [inline lib/config.js]: this means that npm install events is needed?  Also should it be "require('events').EventEmitter"?

**Comment by atodd** [comment]
I've submitted a pull request that also addresses this issue https://github.com/etsy/statsd/pull/637.  It removes the need for `util.inherits` and uses the constructor approach instead.  I also added node versions 6 and 8 to the travis ci config.  All tests are currently passing.  I'm happy to help where needed.  This is an extremely usefully tool and I'd like to see it keep up with current node versions.

File: lib/config.js (hunk 1 above)
Comment (lapo-luchini) [reply lib/config.js]: I guess it should; and that's how it's done in #637.

**Pull request merged**